library/src/interactor_impl.cxx

```diff
@@ -78,158 +78,148 @@ class interactor_impl::internals
+        render = true;
         break;
       case 'V':
-        if (renWithColor)
-        {
-          renWithColor->SetUseVolume(!renWithColor->UsingVolume());
-          log::debug(renWithColor->GetRenderingDescription());
-          renWin->Render();
-        }
+        self->Options.toggle("volume");
+        render = true;
         break;
       case 'I':
-        if (renWithColor)
-        {
-          renWithColor->SetUseInverseOpacityFunction(!renWithColor->UsingInverseOpacityFunction());
-          renWin->Render();
-        }
+        self->Options.toggle("inverse");
+        render = true;
         break;
       case 'O':
-        if (renWithColor)
-        {
-          renWithColor->SetUsePointSprites(!renWithColor->UsingPointSprites());
-          log::debug(renWithColor->GetRenderingDescription());
-          renWin->Render();
-        }
+        self->Options.toggle("point-sprites");
+        render = true;
         break;
       case 'F':
-        if (!renWin->GetFullScreen())
-        {
-          // save current window position and size
-          int* pos = renWin->GetPosition();
-          self->WindowPos[0] = pos[0];
-          self->WindowPos[1] = pos[1];
-          int* size = renWin->GetSize();
-          self->WindowSize[0] = size[0];
-          self->WindowSize[1] = size[1];
-        }
-
-        renWin->SetFullScreen(!renWin->GetFullScreen());
-
-        if (!renWin->GetFullScreen() && self->WindowSize[0] > 0)
-        {
-          // restore previous window position and size
-          renWin->SetPosition(self->WindowPos);
-          renWin->SetSize(self->WindowSize);
-        }
+        self->Options.toggle("fullscreen");
+        render = true;
+        /*        if (!renWin->GetFullScreen())
+                {
+                  // save current window position and size
+                  int* pos = renWin->GetPosition();
+                  self->WindowPos[0] = pos[0];
+                  self->WindowPos[1] = pos[1];
+                  int* size = renWin->GetSize();
+                  self->WindowSize[0] = size[0];
+                  self->WindowSize[1] = size[1];
+                }
+
+                renWin->SetFullScreen(!renWin->GetFullScreen());
+
+                if (!renWin->GetFullScreen() && self->WindowSize[0] > 0)
+                {
+                  // restore previous window position and size
+                  renWin->SetPosition(self->WindowPos);
+                  renWin->SetSize(self->WindowSize);
+                }
+
+                // when going full screen, the OpenGL context changes, we need to reinitialize
+                // the interactor, the render passes and the grid actor.
+                ren->ShowGrid(ren->IsGridVisible());
+                ren->SetupRenderPasses();
+                rwi->ReInitialize();
+
+                renWin->Render();*/ // TODO currently this breaks.
+        // should we just remove this hotkey as it is broken and nobody cares much about it ?
```

Comment:
I think we may want to remove this hotkey or even the option, wdyt @Meakk

Reply:
As it is completely broken

Reply:
I agree. It should be handled by the window manager anyway